### DISCUSSION TITLE
Span entire terminal width

### DISCUSSION THREAD

**Discussion by vincentfiestada** [charmbracelet/lipgloss#178] [General]
Is there a way for lipgloss to style text so that it spans the entire width of the terminal? I would like to render some text so that it's right-aligned on the terminal.

**Comment by dploeger** [comment]
The only way I found yet was to fetch a WindowSizeMsg (if using Lipgloss with Bubbletea, ymmv) and set the width accordingly.

**Comment by muesli** [comment]
You can retrieve the dimensions of the terminal using https://pkg.go.dev/golang.org/x/term#GetSize and supply these values to lipgloss's `Width` and `Height` methods.

**Reply by dploeger** [reply]
You just need to make sure, that they're changed when the terminal is resized.

**Reply by muesli** [reply]
If you're writing a proper TUI, yes. In that case (and if you're using Bubble Tea) you indeed want to adjust your model to listen for `WindowSizeMsg`s being emitted.

**Reply by vincentfiestada** [reply]
Thanks. For some reason using `term.GetSize` inside a PowerShell prompt causes an error, but I think it has more to do with how PowerShell draws its prompt. I ended up passing `$Host.UI.RawUI.WindowSize.Width` as an argument instead.
